source/ada/lsp-ada_contexts.adb

```diff
@@ -814,6 +766,67 @@ package body LSP.Ada_Contexts is
       end if;
    end Get_Node_At;
 
+   --------------------
+   -- Get_PP_Options --
+   --------------------
+
+   procedure Get_PP_Options
+     (Self       : Context;
+      Project    : GNATCOLL.Projects.Project_Type;
+      File       : GNATCOLL.VFS.Virtual_File;
+      PP_Options : in out Utils.Command_Lines.Command_Line)
+   is
+      use type GNAT.Strings.String_Access;
+
+      Options   : GNAT.Strings.String_List_Access;
+      Validated : GNAT.Strings.String_List_Access;
+      Last      : Integer;
+      Default   : Boolean;
+
+   begin
+      Project.Switches
+        (In_Pkg           => "Pretty_Printer",
+         File             => File,
+         Language         => "ada",
+         Value            => Options,
+         Is_Default_Value => Default);
+
+      --  Initialize an gnatpp command line object
+      Last := Options'First - 1;
+      for Item of Options.all loop
```

Comment:
Lacking comments. Put a comment to say that we are counting valid options here for instance, people don't want to analyze each loop in your code